### ISSUE TITLE
Saga III: Another Original Play by a Computer

### ISSUE THREAD

**Issue by lizadaly** [dariusk/NaNoGenMo-2015#116]
Inspired by one of the earliest computer generated narratives: SAGA II, created by Douglas T. Ross and Harrison R. Morse at MIT and featured on _The Thinking Machine_, a 1962 television special co-produced by CBS and MIT. The entire hour-long special can be <a href="http://video.mit.edu/watch/the-thinking-machine-1961-mit-centennial-film-6712/">viewed at MIT</a>.

I came upon the work the day before NaNoGenMo 2015 started (!) while taking Mark Sample's EdX course on <a href="https://www.edx.org/course/electronic-literature-davidsonx-d004x">Electronic Literature</a>, where it's mentioned in a lecture!

<a href="https://github.com/lizadaly/nanogenmo2015">Source code and more research and implementation details</a>.
## Sample

```
 The gun is in the robber's right hand. The money is in the robber's
 left hand. The holster is on the robber. The sheriff's gun is in the
  sheriff's right hand. The sheriff's holster is on the sheriff. The
          glass is on the table. The bottle is on the table.

                                ROBBER
      (The robber is at the window.) Open door; go through door;
  close door; go to corner; put money on corner; go to table; go to
 window; check gun; go to corner; go to table; pick up the glass with
   the robber's left hand; go to window; go to corner; count money

                               SHERIFF
               Go to window; open door; go through door

                                ROBBER
                         Fire; sheriff NICKED

                               SHERIFF
                           Close door; aim

                                ROBBER
                             Fire; MISSED

                               SHERIFF
                             Fire; MISSED

                                ROBBER
                         Fire; sheriff NICKED

                               SHERIFF
                       Aim; fire; robber NICKED

                                ROBBER
          Aim; fire; sheriff HIT; aim; fire; sheriff NICKED

                               SHERIFF
                            Sheriff dies.

                                ROBBER
   Blow out barrel; put gun in holster; pick up the money with the
 robber's right hand; go to table; open door; go through door; close
                                 door

                               CURTAIN
```

<a href="https://raw.githubusercontent.com/lizadaly/nanogenmo2015/master/saga-3.txt">Raw text file of completed "play"</a>.

**Comment by hugovk** [comment]
Brilliant!

([Here's a few other screenshots](https://www.flickr.com/photos/hugovk/albums/72157660170545382) from _The Thinking Machine_).

**Comment by cpressey** [comment]
Oh, that's possibly the program that I thought "had an exciting name like STORY or STORYTELLER" [here](https://github.com/dariusk/NaNoGenMo/issues/39#issuecomment-28069868).  The transcript looks similar to what I dimly remember dimly recalling when I wrote that comment, anyway.

**Comment by lizadaly** [comment]
Sounds like it. I had to make some guesses about the world rules in the original. Reading the memorandum suggests that it's all weighted probabilities and not much (if any) strict rules, which leads to a pretty loose vibe. From one of the 1961 screenplays:

```
SHERIFF: (The sheriff is at the window.) See robber; 
(robber sees sheriff); go to door.

ROBBER: Take gun from holster with right hand; check gun; 
go to door; check gun; put gun down at door.
```

The Robber "wants" to have the gun and shoot the sheriff, but for some inexplicable reason here he puts it down (and then has to pick it up in a subsequent turn).

I considered writing this in Inform, but decided that would make the transcript too tidy as the world models are so thoroughly baked in. I wanted it to be more random, like the original.

**Comment by hugovk** [comment]
Well, in _The Thinking Machine_ programme, there's that bit towards the end when they made one of the glitchy scripts.

**Comment by lizadaly** [comment]
Yeah, that was a fascinating New Aesthetic-esque flash forward. It was hilarious how much it looked like a modern 3D game glitch video.

**Comment by greg-kennedy** [comment]
Looks somewhat similar to what I'm doing - was even thinking of doing a "play" output type, because it makes sense as a way to present a bunch of actions like you'd get from a text adventure or whatever.
